### ISSUE TITLE
Implementation of reactions

### ISSUE THREAD

**Issue by JonasMuehlmann** [JonasMuehlmann/BIB_VPR#113]
Concept:
![image](https://user-images.githubusercontent.com/49788305/124962931-7e269d80-e01f-11eb-910a-933dbded8bb9.png)

- Users are able to react with emojies
~~- Reactions are included in the message(map of reaction to count)~~
DB schema:
```sql
CREATE TABLE Reactions(
Id INT IDENTITY(1,1),
MessageId INT FOREIGN KEY REFERENCES Messages(messageId) NOT NULL,
Reaction NVARCHAR(8) NOT NULL,
Count INT NOT NULL
)
```
```sql
CREATE OR ALTER PROCEDURE AddOrUpdateReaction @messageId INT, @reaction NVARCHAR(8)
AS
BEGIN
	IF  Exists(SELECT * FROM Reactions WHERE Reaction = @reaction AND messageID = @messageId)
	BEGIN
	UPDATE
		REACTIONS
	SET
		Count = Count + 1
		WHERE
		        messageID = @messageId
			AND
			reaction = @reaction;

		SELECT
			Id
		FROM
			Reactions
		WHERE
			messageID = @messageId
			AND
			reaction = @reaction;
	END
	ELSE
	BEGIN
		INSERT INTO
			Reactions
		VALUES(@messageId, @reaction, 1);

		SELECT SCOPE_IDENTITY();
	END;
END;
```
```sql
CREATE OR ALTER PROCEDURE RemoveOrUpdateReaction @messageId INT, @reaction NVARCHAR(8)
AS
BEGIN
	IF  (SELECT r.Count FROM Reactions r WHERE Reaction = @reaction AND messageID = @messageId) = 1
	BEGIN
		DELETE FROM
			Reactions
		WHERE
			messageID = @messageId
			AND
			reaction = @reaction;
	END
	ELSE
	BEGIN
		UPDATE
			REACTIONS
		SET
			Count = Count - 1
			WHERE
				Reaction = @reaction;
		END
END;

```
Backend tasks:

- [x] `bool Message.AddReaction(uint messageId, string reaction)`
- [x] `bool Message.RemoveReaction(uint messageId, string reaction)`
- [x] `Dictionary<string, int> Message.RetrieveReactions(uint messageId)`
- [x] Integrate with SignalR

Frontend tasks:

- [ ] UI controls for adding reactions
- [ ] List emojis in grid
- [ ] Filter  emojis via search
- [ ] Filter emojis via categories
- [x] Displaying reactions

Help wanted on:
- Frontend tasks

**Comment by JayKim94** [comment]
Frontend tasks will be started as soon as #262 gets merged

**Comment by JayKim94** [comment]
next issue moved to #343

**Comment by JayKim94** [comment]
Send Emojis with Emoji -picker: closed by #262
